pkg/services/apiserver/builder/common.go

```diff
@@ -43,11 +42,11 @@ type APIGroupBuilder interface {
 	// Standard namespace checking will happen before this is called, specifically
 	// the namespace must matches an org|stack that the user belongs to
 	GetAuthorizer() authorizer.Authorizer
+}
 
-	// Get the desired dual writing mode. These are modes 1, 2, 3 and 4 if
-	// the feature flag `unifiedStorage` is enabled and mode 0 if it is not enabled.
-	// #TODO add type for map[string]grafanarest.DualWriterMode?
-	GetDesiredDualWriterMode(dualWrite bool, toMode map[string]grafanarest.DualWriterMode) grafanarest.DualWriterMode
+// When implemented (and enabled), this will override any user defined setting
+type DualWriteModeOverrider interface {
```

Comment:
rather than make everyone implement this function, we can have it optional and provide good defaults when not configured